File: src/test/scala/org/egonet/graph/GraphTest.scala
package org.egonet.graph

import java.util.Random
import scala.math.abs
import org.egonet.EgonetSpec

class GraphTest extends EgonetSpec {

  val rand = new Random();
  
  	"testRandom" should "have correct size and density" in {
  	  
  	  for(i <- Range(0, 60)) {
			var expectedDensity = rand.nextDouble();
			
			var graph = Graph.random(i, expectedDensity);
			assert(i.intValue() == graph.nodes().size());
			
			var calculatedDensity = graph.density()
			if(i > 50) {
			  var diff = abs(calculatedDensity - expectedDensity)
			  
			  if(diff > 0.1)
			    assert(expectedDensity == calculatedDensity)
			}
  	    
  	  }
	}
  
}

File: src/test/java/org/egonet/graph/GraphTest.java
package org.egonet.graph;

import static org.junit.Assert.*;

import java.util.Random;

import org.egonet.graph.Graph;
import org.junit.*;
import junit.framework.JUnit4TestAdapter;

public class GraphTest {

	private Random rand = new Random();
	
	@Test
	public void testRandom() {
		for(Integer i = 0; i < 60; i++) {
			Double expectedDensity = rand.nextDouble();
			Graph graph = Graph.random(i, expectedDensity);
			
			assertEquals("Random graph of "+i+" nodes actually has "+i+" nodes.",
					i.intValue(),graph.nodes().size());
			Double calculatedDensity = graph.density();
			if(i > 50) {
				assertEquals("Random graph with "+i+" nodes has requested density.",
						expectedDensity,calculatedDensity,0.1);
			}
		}
	}

	public static junit.framework.Test suite() {
      return new JUnit4TestAdapter(GraphTest.class);
	}
}
